src/mam/mod_mam_muc_rdbms_arch.erl

```diff
@@ -334,6 +334,18 @@ lookup_messages(Host, RoomID, RoomJID = #jid{},
     {ok, {TotalCount, Offset,
           rows_to_uniform_format(MessageRows, Host, RoomJID)}}.
 
+-spec get_mam_muc_gdpr_data(jid:username(), jid:server()) -> {ok, mod_mam:messages()}.
+get_mam_muc_gdpr_data(Username, LServer) ->
+    LUser = jid:nodeprep(Username),
+    WithJID = jid:to_binary({LUser, LServer}),
+    %% TODO What is PageSize ?
+    SWithNick = mongoose_rdbms:escape_string(WithJID),
+    Filter = make_filter_nickname_only(SWithNick),
+    Rows = extract_messages(LServer, Filter, 0, 50, false),
```

Comment:
Will it work for more than 50 messages? 
You set `IMax` to 50 therefore I think it will return no more than 50 messages.

Reply:
yes, thats the reason I put `TODO` comment - to investigate what should be the value